homeassistant/components/deconz/config_flow.py

```diff
@@ -182,27 +178,18 @@ def _update_entry(self, entry, host, port, api_key=None):
         if discovery_info[ssdp.ATTR_UPNP_MANUFACTURER_URL] != DECONZ_MANUFACTURERURL:
             return self.async_abort(reason="not_deconz_bridge")
 
-        uuid = discovery_info[ssdp.ATTR_UPNP_UDN].replace("uuid:", "")
-
-        _LOGGER.debug("deCONZ gateway discovered (%s)", uuid)
+        self.bridgeid = normalize_bridge_id(discovery_info[ssdp.ATTR_UPNP_SERIAL])
+        await self.async_set_unique_id(self.bridgeid)
 
         parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
 
         for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if uuid == entry.data.get(CONF_UUID):
-                if entry.source == "hassio":
-                    return self.async_abort(reason="already_configured")
+            if self.bridgeid == entry.unique_id and entry.source != "hassio":
                 return self._update_entry(entry, parsed_url.hostname, parsed_url.port)
 
-        bridgeid = discovery_info[ssdp.ATTR_UPNP_SERIAL]
-        if any(
-            bridgeid == flow["context"][CONF_BRIDGEID]
-            for flow in self._async_in_progress()
-        ):
-            return self.async_abort(reason="already_in_progress")
+        self._abort_if_unique_id_configured()
```

Comment:
Why not put this in the for loop above? It's just doing an abort `if self.bridgeid == entry.unique_id`